homeassistant/components/vizio/config_flow.py

```diff
@@ -0,0 +1,169 @@
+"""Config flow for Vizio."""
+import logging
+from typing import Any, Dict
+
+from pyvizio import VizioAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_DEVICE_CLASS,
+    CONF_HOST,
+    CONF_NAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
+
+from . import validate_auth
+from .const import (
+    CONF_UNIQUE_ID,
+    CONF_VOLUME_STEP,
+    DEFAULT_DEVICE_CLASS,
+    DEFAULT_NAME,
+    DEFAULT_VOLUME_STEP,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def update_schema_defaults(input_dict: Dict[str, Any]) -> vol.Schema:
+    """Update schema defaults based on user input/config dict. Retains info already provided for future form views."""
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=input_dict.get(CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(CONF_HOST, default=input_dict.get(CONF_HOST)): str,
+            vol.Optional(
+                CONF_DEVICE_CLASS,
+                default=input_dict.get(CONF_DEVICE_CLASS, DEFAULT_DEVICE_CLASS),
+            ): vol.All(str, vol.Lower, vol.In(["tv", "soundbar"])),
+            vol.Optional(
+                CONF_ACCESS_TOKEN, default=input_dict.get(CONF_ACCESS_TOKEN, "")
+            ): str,
+        }
+    )
```

Comment:
Set extra keyword argument to remove extra, otherwise the import config may fail to validate.